### PR TITLE
SNYK container scan integrated into build workflow

### PR DESCRIPTION
**Pull request by prabhakarreddy1234**
Added SNYK container scanning step  in build workflow for  auth server and test client images.

[x] Rebase to master
[x] Attach to trello
[x] Clean commit history

### CODE REVIEW

**Comment by prabhakarreddy1234** [comment]
[get an identity -  SNYK  scan integration](https://trello.com/c/JoLARToB/685-get-an-identity-snyk-scan-integration)

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -21,15 +21,17 @@ jobs:
     runs-on: ubuntu-latest
 
     steps:
-    - uses: styfle/cancel-workflow-action@0.9.1
-      with:
-        access_token: ${{ github.token }}
-      if: github.event_name == 'pull_request'
+      - uses: styfle/cancel-workflow-action@0.9.1
```
Comment (merrilmathew) [inline .github/workflows/build.yml:24]: is indentation right?

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -176,16 +224,26 @@ jobs:
...
+      - uses: ./.github/workflows/actions/deploy-environment
+        id: deploy
+        with:
+          environment_name: dev
+          image_tag: ${{ github.sha }}
+          azure_credentials: ${{ secrets.AZURE_CREDENTIALS }}
+          terraform_vars: workspace_variables/dev.tfvars.json
+          terraform_backend_vars: workspace_variables/dev.backend.tfvars
+
+      - name: Check the deployed service URL
```
Comment (merrilmathew) [inline .github/workflows/build.yml:238]: please pull the latest main. This task is no longer there.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -110,28 +112,62 @@ jobs:
...
+      run: docker image push ${{ steps.image_tags.outputs.authserver }}
+
+    - name: Test client docker build
+      uses: docker/build-push-action@v3
       with:
         context: dotnet-authserver/src/TeacherIdentity.TestClient
         push: true
         tags: ${{ steps.image_tags.outputs.testclient }}
+        build-args: |
+          GIT_SHA=${{ github.sha }}
```
Comment (gunndabad) [inline .github/workflows/build.yml:170]: Don't need this arg here; the `Dockerfile` for test client won't use it

**Review by saliceti — APPROVED** [review]

**Pull request merged**